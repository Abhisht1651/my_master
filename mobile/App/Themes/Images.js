// leave off @2x/@3x
const newzeraWhiteLogo = require('../Images/newzera_logo_v_white.png')

// Poll Icons
const pollActiveIcon = require('../Images/Icons/icon-active/icon-active.png')
const pollInactiveIcon = require('../Images/Icons/icon-inactive/icon-inactive.png')
const voterIcon1 = require('../Images/Icons/voters/1/Bitmap.png')
const voterIcon2 = require('../Images/Icons/voters/2/Bitmap.png')
const voterIcon3 = require('../Images/Icons/voters/3/Bitmap.png')
const shareIcon = require('../Images/Icons/icon-action-share/icon-action-share.png')
const ravenIcon = require('../Images/Icons/icon-action-raven/icon-action-raven.png')
const reactIcon = require('../Images/Icons/icon-action-react/icon-action-react.png')
const menuIcon = require('../Images/Icons/icon-poll-big/icon-poll-big.png')
const commentIcon = require('../Images/Icons/icon-story-comments/icon-story-comments.png')
const iconPollDark = require('../Images/Icons/icon-poll-dark.png')

// StoryLine
const iconStoryLineTrending = require('../Images/Icons/icon-trending/icon-trending.png')
const newsImage = require('../Images/NewsImage.png')
const storyImage = require('../Images/StoryLineImage.png')
const iconStoryLineReach = require('../Images/Icons/icon-reach/icon-reach.png')
const iconPeople = require('../Images/Icons/icon-people/icon-people.png')
const followButton = require('../Images/Button-small-outline/Button-small-outline.png')
const iconStoryLineInactive = require('../Images/Icons/icon-inactive/icon-inactive.png')
const iconStoryLineRelatedEntity1 = require('../Images/Icons/StoryLineRelatedEntities/1/Bitmap.png')
const iconStoryLineRelatedEntity2 = require('../Images/Icons/StoryLineRelatedEntities/2/Bitmap.png')
const iconStoryLineRelatedEntity3 = require('../Images/Icons/StoryLineRelatedEntities/3/Bitmap.png')
const iconStoryLineRelatedEntity4 = require('../Images/Icons/StoryLineRelatedEntities/4/Bitmap.png')
const sourceIcon = require('../Images/Source/Source.png')
const majorEventsNotSelected = require('../Images/Icons/icon-major-events-not-selected.png')
const majorEventsSelected = require('../Images/Icons/icon-major-events-selected.png')
const allEventsNotSelected = require('../Images/Icons/icon-all-events-not-selected.png')
const allEventsSelected = require('../Images/Icons/icon-all-events-selected.png')
const player = require('../Images/Player/Player.png')
const chevron = require('../Images/Icons/chevron.png')
const iconSource = require('../Images/Icons/icon-source.png')

// Bottom Nav
const newsfeed = require('../Images/newsfeed.png')
const newsfeed_sel = require('../Images/newsfeed-selected.png')
const search = require('../Images/search.png')
const search_sel = require('../Images/search-selected.png')
const profile = require('../Images/profile.png')
const profile_sel = require('../Images/profile-selected.png')
const chat = require('../Images/chat.png')
const chat_sel = require('../Images/chat-selected.png')
const friendsactivity = require('../Images/friendsactivity.png')
const friendsactivity_sel = require('../Images/friendsactivity-selected.png')

const faq = require('../Images/Icons/faq-icon.png')
const closeButton = require('../Images/Icons/close-button.png')
const newzeraPurpleLogo = require('../Images/newzera_logo_v_purple.png')
const cross = require('../Images/cross.png')
const leftArrow = require('../Images/Icons/back/back.png')
const defaultProfile = require('../Images/Icons/default-profile-image.png')
const profileCamera = require('../Images/Icons/icon-camera-outline.png')
const locationIcon = require('../Images/Icons/location-icon.png')
const importContacts = require('../Images/Icons/importContacts.png')
const menuIconNotification = require('../Images/Icons/icon-menu-notifiction.png')
const newzeraPurpleLogoH = require('../Images/newzeraPurpleLogoH.png')
const notificationIcon = require('../Images/Icons/icon-notifications.png')
const bitmap = require('../Images/Icons/Bitmap.png')
const followStorylines = require('../Images/Icons/icon-followedstorylines.png')
const followEntities = require('../Images/Icons/icon-followedentities.png')
const myPolls = require('../Images/Icons/icon-poll.png')
const friends = require('../Images/Icons/icon-friends.png')
const friendRequests = require('../Images/Icons/icon-requests.png')
const inviteFriends = require('../Images/Icons/icon-friend-add.png')
const settings = require('../Images/Icons/icon-settings.png')
const nightModeUnactive = require('../Images/Icons/icon-nightmode-unactive.png')
const checkMark = require('../Images/Icons/icon-checkmark-purple.png')

// List of friends
const picture = require('../Images/Story2x.png')

// Stories from friends
const profilePicture1 = require('../Images/Bitmap2x.png')
const flagPicture = require('../Images/Bitmap.png')
const profilePicture2 = require('../Images/Bitmap1.png')
const profilePicture3 = require('../Images/Profile2.png')
const profilePicture4 = require('../Images/Profile3.png')

// Profile pictures
const EmptyProfilePicture = require('../Images/ProfilePicture.png')
const bitmap2x = require('../Images/Bitmap2x.png')
const camera2x = require('../Images/icon-camera2x.png')
const profilePicture = require('../Images/Profilepicture2x.png')
const profilePictureSmall = require('../Images/ManSmiling.png')

// List of users
const listOfUsersImage = require('../Images/Group.png')

// Pop ups
const closeIconPurple = require('../Images/icon-close-purple2x.png')
const currentLocationIcon = require('../Images/icon-currentlocation.png')
const world = require('../Images/icon-world2x.png')
const national = require('../Images/icon-national2x.png')
const science = require('../Images/icon-science2x.png')
const sports = require('../Images/icon-sports2x.png')
const technology = require('../Images/icon-technology2x.png')
const entertainment = require('../Images/icon-entretainment2x.png')
const others = require('../Images/icon-others2x.png')
const closeIcon = require('../Images/Icons/icon-close-navy/icon-close-navy2x.png')

// SearchSuggestion
const currentLocationIconSmall = require('../Images/Icons/icon-currentlocation-small.png')

// Time Filter
// const closeIcon = require('../Images/Icons/icon-close-navy/icon-close-navy.png')
const checkmark = require('../Images/Icons/icon-checkmark-purple/icon-checkmark-purple.png')

// Activity Feed
const profile1 = require('../Images/profile1-2x.png')
const profile2 = require('../Images/profile2-2x.png')
const follow1 = require('../Images/follow1-2x.png')
const elipsis = require('../Images/elipsis.png')
const filtericon = require('../Images/filtericon.png')
const iconcopypoll = require('../Images/iconcopypoll.png')
const iconreact = require('../Images/iconreact.png')
const iconcomment = require('../Images/iconcomment.png')
const icontrending = require('../Images/icontrending.png')
const iconcloseblue = require('../Images/icon_close_blue.png')
const checkedmark = require('../Images/checkedmark.png')

// Entities Component
const EntitiesImage0 = require('../Images/Entities/0/Bitmap.png')
const EntitiesImage1 = require('../Images/Entities/1/Bitmap-1.png')
const EntitiesImage2 = require('../Images/Entities/2/Bitmap-2.png')
const EntitiesImage3 = require('../Images/Entities/3/Bitmap-3.png')
const EntitiesImage4 = require('../Images/Entities/4/Bitmap-4.png')
const EntitiesImage5 = require('../Images/Entities/5/Bitmap-5.png')
const EntitiesImage6 = require('../Images/Entities/6/Bitmap-6.png')
const EntitiesImage7 = require('../Images/Entities/7/Bitmap-7.png')
const EntitiesImage8 = require('../Images/Entities/8/Bitmap-8.png')
const EntitiesImage9 = require('../Images/Entities/9/Bitmap-9.png')
const EntitiesImage10 = require('../Images/Entities/10/Bitmap-10.png')
const EntitiesImage11 = require('../Images/Entities/11/Bitmap-11.png')

// Back Button
const arrow = require('../Images/Arrow.png')
// Comments
const replyIcon = require('../Images/Icons/icon-reply/icon-reply.png')
const boostIcon = require('../Images/Icons/icon-boost/icon-boost.png')

// Poll Screen
const backIcon = require('../Images/Icons/back/back.png')
const iconStoryLineBig = require('../Images/Icons/icon-storyline-big.png')

// Chats
const write = require('../Images/Icons/write.png')
const right_purple = require('../Images/right_purple.png')
const ss1 = require('../Images/ss1.png')
const ss2 = require('../Images/ss2.png')
const ss3 = require('../Images/ss3.png')
const ss4 = require('../Images/ss4.png')
const ss5 = require('../Images/ss5.png')
const ss6 = require('../Images/ss6.png')
const iconelipsis = require('../Images/icon_elipsis.png')
// Raven Flow
const light = require('../Images/group_2.png')
const lightfill = require('../Images/lightfilled.png')
const background_sent = require('../Images/background.png')
const right = require('../Images/right.png')
const arrowleft = require('../Images/arrow_left.png')
const overimage = require('../Images/overimage.png')

// NewsFeed
const logopurple = require('../Images/NE_logo_purple_2x.png')
const iconmenu = require('../Images/iconmenu.png')

// Discover
const iconClose = require('../Images/Icons/icon-close-small.png')

// Notifications
const iconTrending = require('../Images/Icons/IconTrending.png')
const iconList = require('../Images/Icons/IconList.png')
const boost = require('../Images/Icons/Boost.png')
const dots3 = require('../Images/Icons/Dots3.png')
const iconFollowedentities = require('../Images/Icons/iconFollowedentities.png')
const iconFollowedstorylines = require('../Images/Icons/iconFollowedstorylines.png')
const iconFriends = require('../Images/Icons/iconFriends.png')
const iconFriendsInvite = require('../Images/Icons/iconFriendsInvite.png')
const iconNotify = require('../Images/Icons/IconNotify.png')
const iconPoll = require('../Images/Icons/iconPoll.png')
const iconRequests = require('../Images/Icons/iconRequests.png')
const leftArrowback = require('../Images/Icons/LeftArrow.png')
const rightSmallArrow = require('../Images/Icons/RightSmallArrow.png')
const msg = require('../Images/Icons/Msg.png')
const raven = require('../Images/Icons/Raven.png')
const iconSettings = require('../Images/Icons/iconSettings.png')
const iconNightmode = require('../Images/Icons/iconNightmode.png')
const iconClose1 = require('../Images/Icons/iconClose.png')
const iconImport = require('../Images/Icons/iconImport.png')
const amanda = require('../Images/People/Amanda.png')
const sean = require('../Images/People/Sean.png')
const trump = require('../Images/People/Trump.png')
const error = require('../Images/Error.png')

const images = {
  newzeraWhiteLogo,
  pollActiveIcon,
  pollInactiveIcon,
  voterIcon1,
  voterIcon2,
  voterIcon3,
  shareIcon,
  ravenIcon,
  reactIcon,
  menuIcon,
  commentIcon,
  newsfeed,
  newsfeed_sel,
  search,
  search_sel,
  profile,
  profile_sel,
  chat,
  chat_sel,
  friendsactivity,
  friendsactivity_sel,
  newsImage,
  storyImage,
  iconStoryLineTrending,
  iconStoryLineReach,
  iconPeople,
  followButton,
  faq,
  closeButton,
  newzeraPurpleLogo,
  cross,
  leftArrow,
  defaultProfile,
  profileCamera,
  locationIcon,
  importContacts,
  menuIconNotification,
  newzeraPurpleLogoH,
  notificationIcon,
  bitmap,
  followStorylines,
  followEntities,
  myPolls,
  friends,
  friendRequests,
  inviteFriends,
  settings,
  nightModeUnactive,
  checkMark,
  iconStoryLineInactive,
  iconStoryLineRelatedEntity1,
  iconStoryLineRelatedEntity2,
  iconStoryLineRelatedEntity3,
  iconStoryLineRelatedEntity4,
  sourceIcon,
  picture,
  profilePicture1,
  profilePicture2,
  profilePicture3,
  profilePicture4,
  flagPicture,
  EmptyProfilePicture,
  bitmap2x,
  camera2x,
  profilePicture,
  profilePictureSmall,
  listOfUsersImage,
  closeIconPurple,
  currentLocationIcon,
  world,
  national,
  science,
  sports,
  technology,
  entertainment,
  checkmark,
  others,
  closeIcon,
  EntitiesImage0,
  EntitiesImage1,
  EntitiesImage2,
  EntitiesImage3,
  EntitiesImage4,
  EntitiesImage5,
  EntitiesImage6,
  EntitiesImage7,
  EntitiesImage8,
  EntitiesImage9,
  EntitiesImage10,
  EntitiesImage11,
  arrow,
  replyIcon,
  boostIcon,
  backIcon,
  currentLocationIconSmall,
  majorEventsNotSelected,
  majorEventsSelected,
  allEventsNotSelected,
  allEventsSelected,
  player,
  chevron,
  iconSource,
  write,
  right_purple,
  ss1,
  ss2,
  ss3,
  ss4,
  ss5,
  ss6,
  iconelipsis,
  iconStoryLineBig,
  logopurple,
  filtericon,
  iconmenu,
  profile1,
  profile2,
  follow1,
  elipsis,
  iconcopypoll,
  iconreact,
  icontrending,
  iconcomment,
  iconcloseblue,
  checkedmark,
  light,
  lightfill,
  background_sent,
  right,
  arrowleft,
  overimage,
  iconClose,
  iconPollDark,
  iconTrending,
  iconList,
  boost,
  dots3,
  iconFollowedentities,
  iconFollowedstorylines,
  iconFriends,
  iconFriendsInvite,
  iconNotify,
  iconPoll,
  iconRequests,
  leftArrowback,
  rightSmallArrow,
  msg,
  raven,
  iconSettings,
  iconNightmode,
  iconClose1,
  iconImport,
  amanda,
  sean,
  trump,
  error
}

export default images
