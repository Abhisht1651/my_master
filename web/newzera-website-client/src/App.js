// @flow

import React from 'react'

// import NewzeraWebsite from './NewzeraWebsite'
// import LoginFlow from './LoginFlow'
// import Newsfeed from './Newsfeed'
// import UserProfile from './UserProfile'
import Notifications from './Notifications'

function App() {
  return (
    <div>
      <Notifications />
    </div>
  )
}

export default App
