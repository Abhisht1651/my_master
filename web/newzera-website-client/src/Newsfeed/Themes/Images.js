const John = 'assests/newsfeed/John.jpg'
const NewzeraLogo = 'assests/newsfeed/NewzeraLogo.jpg'
const DownArrow = 'assests/newsfeed/DownArrow.jpg'
const NewsFeed = 'assests/newsfeed/NewsFeed.jpg'
const FollowedStorylines = 'assests/newsfeed/FollowedStorylines.jpg'
const FollowedEntities = 'assests/newsfeed/FollowedEntities.jpg'
const Friends = 'assests/newsfeed/Friends.jpg'
const FriendsActivity = 'assests/newsfeed/FriendsActivity.jpg'
const Raven = 'assests/newsfeed/Raven.jpg'
const Settings = 'assests/newsfeed/Settings.jpg'
const NightMode = 'assests/newsfeed/NightMode.jpg'
const Search = 'assests/newsfeed/Search.jpg'
const Notifications = 'assests/newsfeed/Notifications.jpg'
const Trending = 'assests/newsfeed/Trending.jpg'
const Reach = 'assests/newsfeed/Reach.jpg'
const Demo1 = 'assests/newsfeed/Demo1.jpg'
const India = 'assests/newsfeed/India.jpg'
const Message = 'assests/newsfeed/Message.jpg'
const Demo2 = 'assests/newsfeed/Demo2.jpg'
const Demo3 = 'assests/newsfeed/Demo3.jpg'
const CategoryFilter = 'assests/newsfeed/CategoryFilter.jpg'
const StorylineActive = 'assests/newsfeed/StorylineActive.jpg'
const People = 'assests/newsfeed/People.jpg'
const Share = 'assests/newsfeed/Share.jpg'
const React = 'assests/newsfeed/React.jpg'
const Demo4 = 'assests/newsfeed/Demo4.jpg'
const PollEnded = 'assests/newsfeed/PollEnded.jpg'
const Comment = 'assests/newsfeed/Comment.jpg'
const Poll = 'assests/newsfeed/Poll.jpg'

const Cross = 'assests/CategoryFilter/Cross.jpg'
const Entertainment = 'assests/CategoryFilter/Entertainment.jpg'
const National = 'assests/CategoryFilter/National.jpg'
const Science = 'assests/CategoryFilter/Science.jpg'
const Sports = 'assests/CategoryFilter/Sports.jpg'
const Technology = 'assests/CategoryFilter/Technology.jpg'
const World = 'assests/CategoryFilter/World.jpg'
const Checkmark = 'assests/CategoryFilter/Checkmark.jpg'

const Images = {
  John,
  NewzeraLogo,
  DownArrow,
  NewsFeed,
  FollowedStorylines,
  FollowedEntities,
  Friends,
  FriendsActivity,
  Raven,
  Settings,
  NightMode,
  Search,
  Notifications,
  Trending,
  Reach,
  Demo1,
  India,
  Message,
  Demo2,
  Demo3,
  CategoryFilter,
  StorylineActive,
  People,
  Share,
  React,
  Demo4,
  PollEnded,
  Comment,
  Poll,
  Cross,
  National,
  Sports,
  Science,
  Entertainment,
  Technology,
  World,
  Checkmark
}

export default Images
