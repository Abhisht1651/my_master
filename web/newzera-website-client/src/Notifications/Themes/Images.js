const Notifications = 'assests/Notifications/Notifications.jpg'
const Trending = 'assests/Notifications/Trending.jpg'
const Ellipses = 'assests/Notifications/Ellipses.jpg'
const List = 'assests/Notifications/List.jpg'
const John = 'assests/Notifications/John.jpg'
const Msg = 'assests/Notifications/Msg.jpg'

const Images = {
  Notifications,
  Trending,
  Ellipses,
  List,
  John,
  Msg
}

export default Images
