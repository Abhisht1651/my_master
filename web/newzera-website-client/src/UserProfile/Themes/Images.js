const Rect1 = 'assests/Userfeed/Rect1.jpg'
const AddFriend = 'assests/Userfeed/AddFriend.jpg'
const Ellipses = 'assests/Userfeed/Ellipses.jpg'
const FollowUser = 'assests/Userfeed/FollowUser.jpg'
const John = 'assests/Userfeed/John.jpg'
const DemoUF1 = 'assests/Userfeed/DemoUF1.jpg'
const DemoUF2 = 'assests/Userfeed/DemoUF2.jpg'
const DemoUF3 = 'assests/Userfeed/DemoUF3.jpg'
const DemoUF4 = 'assests/Userfeed/DemoUF4.jpg'
const MsgBlue = 'assests/Userfeed/MsgBlue.jpg'
const Trending = 'assests/Userfeed/Trending.jpg'
const Reach = 'assests/Userfeed/Reach.jpg'
const Comment = 'assests/Userfeed/Comment.jpg'
const Boost = 'assests/Userfeed/Boost.jpg'
const Share = 'assests/Userfeed/Share.jpg'
const StoryShare = 'assests/Userfeed/StoryShare.jpg'

const Images = {
  Rect1,
  AddFriend,
  Ellipses,
  FollowUser,
  John,
  DemoUF1,
  DemoUF2,
  DemoUF3,
  DemoUF4,
  MsgBlue,
  Trending,
  Reach,
  Comment,
  Boost,
  Share,
  StoryShare
}

export default Images
