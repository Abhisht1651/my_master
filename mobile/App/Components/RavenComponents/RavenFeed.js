// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React, { useState } from 'react'
import {
  View,
  Text,
  TouchableWithoutFeedback,
  Image,
  Dimensions,
  TouchableOpacity
} from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/RavenComponentStyles/RavenFeedStyles'
import { Images } from '../../Themes'
import Storyline from '../StoryLineShortCardDefualt'

const Width = Dimensions.get('window').width

type Props = {
  images: Array<Image>,
  friendcount: number,
  friendcount: number,
  sent: Boolean,
  sentimages: Array<Image>,
  ravensentperson: String,
  ravenfriends: PropTypes.func,
  sendmorefriends: PropTypes.func,
  storylineitem: PropTypes.Array,
  ravenimage: Image
} & NavigationContainerProps

const RavenFeed = (props: Props) => {
  const {
    images,
    friendcount,
    sent,
    sentimages,
    ravensentperson,
    ravenfriends,
    sendmorefriends,
    storylineitem,
    ravenimage
  } = props
  const [value, setvalue] = useState(true)
  return (
    <View style={styles.container}>
      <View style={styles.container1}>
        <TouchableWithoutFeedback onPress={ravenfriends}>
          <View style={styles.headercontainer}>
            {images && images.length >= 1 && (
              <Image source={images[0]} style={[styles.image, { left: (16 / 375) * Width }]} />
            )}
            {images && images.length >= 2 && (
              <Image
                source={images[1]}
                style={[styles.image, { left: (16 / 375) * Width + (20 / 375) * Width }]}
              />
            )}
            {images && images.length >= 3 && (
              <Image
                source={images[2]}
                style={[styles.image, { left: (16 / 375) * Width + (40 / 375) * Width }]}
              />
            )}
            {ravenimage && <Image source={ravenimage} style={styles.icon} />}
            <View style={styles.sentravencont}>
              <Text
                style={[
                  styles.sentraventext,
                  {
                    left: images
                      ? (images.length - 1) * (22 / 375) * Width +
                        (28 / 375) * Width +
                        (20 / 375) * Width
                      : (28 / 375) * Width + (20 / 375) * Width
                  }
                ]}>
                {sent ? <Text>You received a raven from </Text> : <Text>You sent a raven to </Text>}
                <Text style={styles.underline}>{friendcount} friends</Text>
              </Text>
            </View>
            <View style={styles.right}>
              <Image source={Images.right} style={styles.righticon} />
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View style={styles.container2}>
        <Storyline
          reach={storylineitem.reach}
          heading={storylineitem.heading}
          storyTitle={storylineitem.storyTitle}
          friendsFollowing={storylineitem.friendsFollowing}
          updatedTime={storylineitem.updatedTime}
          following={storylineitem.following}
          trending={storylineitem.trending}
          storyImage={storylineitem.storyImage}
          border={false}
        />
      </View>
      <View style={styles.container3}>
        <View style={styles.sendmorecont}>
          <TouchableWithoutFeedback onPress={sendmorefriends}>
            <Text style={styles.sendmoretext}>Send to more friends</Text>
          </TouchableWithoutFeedback>
        </View>
      </View>
      {sent && value && (
        <View style={styles.overcontainer}>
          <Image source={Images.overimage} style={styles.imagecontainer} />
          <TouchableOpacity onPress={() => setvalue(false)}>
            <View style={styles.blue}>
              <Image source={Images.lightfill} style={styles.lightfill} />
              <View style={styles.topcontainer}>
                <View style={styles.headercontainer}>
                  {sentimages && sentimages.length >= 1 && (
                    <Image
                      source={sentimages[0]}
                      style={[styles.image, { left: (16 / 375) * Width }]}
                    />
                  )}
                  {sentimages && sentimages.length >= 2 && (
                    <Image
                      source={sentimages[1]}
                      style={[styles.image, { left: (16 / 375) * Width + (20 / 375) * Width }]}
                    />
                  )}
                  {sentimages && sentimages.length >= 3 && (
                    <Image
                      source={sentimages[2]}
                      style={[styles.image, { left: (16 / 375) * Width + (40 / 375) * Width }]}
                    />
                  )}
                  <View style={styles.sentravencont}>
                    <Text
                      style={[
                        styles.overraventext,
                        {
                          left: sentimages
                            ? (sentimages.length - 1) * (22 / 375) * Width +
                              (28 / 375) * Width +
                              (20 / 375) * Width
                            : (28 / 375) * Width + (20 / 375) * Width
                        }
                      ]}>
                      {ravensentperson} sent you a Raven!
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </TouchableOpacity>
        </View>
      )}
    </View>
  )
}

export default RavenFeed
