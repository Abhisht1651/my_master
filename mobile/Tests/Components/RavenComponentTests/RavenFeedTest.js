import React from 'react'
import { shallow } from 'enzyme'
import { TouchableWithoutFeedback } from 'react-native'
import RavenFeed from '../../../App/Components/RavenComponents/RavenFeed'
import Storyline from '../../../App/Components/StoryLineShortCardDefualt'
import { Images } from '../../../App/Themes'
// Test for RavenAll (snapshot and props)
describe('RavenAll', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <RavenFeed
        storylineitem={{
          reach: 45,
          heading: 'World',
          storyTitle: 'UK exit from the European Union',
          friendsFollowing: '45 friends Following',
          updatedTime: '2m',
          following: true,
          trending: true,
          storyImage: Images.storyImage
        }}
      />
    ) // no compile-time error
  })
  it('RavenAll component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('RavenAll BottomNav', () => {
    expect(wrapper.find('BottomNav')).toHaveLength(0)
  })
  it('should render 2 TouchableWithoutFeedback', () => {
    expect(wrapper.find(TouchableWithoutFeedback)).toHaveLength(2)
  })
  it('should render 1 Storyline', () => {
    expect(wrapper.find(Storyline)).toHaveLength(1)
  })
})
