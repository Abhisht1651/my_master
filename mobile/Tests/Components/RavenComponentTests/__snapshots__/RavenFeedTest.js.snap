// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RavenAll RavenAll component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#DCDCE9",
      "borderRadius": 26.0546875,
      "borderWidth": 1.7369791666666665,
      "marginBottom": 21.712239583333336,
      "marginLeft": 50,
      "marginRight": 50,
      "marginTop": 21.712239583333336,
      "width": "86.66%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#DCDCE9",
        "borderBottomWidth": 1.7369791666666665,
        "height": 118.11458333333334,
        "justifyContent": "center",
      }
    }
  >
    <TouchableWithoutFeedback>
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#8F8FAF",
                  "fontFamily": "BananaGrotesk-Regular",
                  "fontSize": 28.000000000000004,
                  "letterSpacing": -0.16,
                  "lineHeight": 32,
                  "width": 400,
                },
                Object {
                  "left": 96,
                },
              ]
            }
          >
            <Text>
              You sent a raven to 
            </Text>
            <Text
              style={
                Object {
                  "textDecorationLine": "underline",
                }
              }
            >
               friends
            </Text>
          </Text>
        </View>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 40,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 24,
                "width": 12,
              }
            }
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#DCDCE9",
        "borderBottomWidth": 1.7369791666666665,
        "height": 644,
      }
    }
  >
    <storyLineShortCardDefault
      border={false}
      followDisplay={true}
      followerDisplay={true}
      following={true}
      friendsFollowing="45 friends Following"
      heading="World"
      onPressFollow={[Function]}
      reach={45}
      storyImage={
        Object {
          "default": "",
        }
      }
      storyTitle="UK exit from the European Union"
      trending={true}
      updatedTime="2m"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 118.11458333333334,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 34.739583333333336,
          "justifyContent": "center",
          "width": 280,
        }
      }
    >
      <TouchableWithoutFeedback>
        <Text
          style={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 28.000000000000004,
              "letterSpacing": -0.16,
              "lineHeight": 32,
              "textAlign": "center",
            }
          }
        >
          Send to more friends
        </Text>
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
